handlers/handlers.go

```diff
@@ -57,6 +65,59 @@ func wrapGetWithErrors(ef func(error), hf handlerFunc) http.Handler {
 	})
 }
 
+func hmacAuth(hf handlerFunc, secretKey string, serviceId string) handlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) (int, error) {
+		// If secret key isn't set, allow all requests
+		if secretKey == "" {
+			return hf(w, r)
+		}
+
+		var rawSignature, rawSignDate string
+
+		query := r.URL.Query()
+
+		if signature := query.Get("signature"); signature != "" {
```

Comment:
I think this could be simplified to:
```
signature := query.get("signature")
if signature == "" {
    signature = r.Header.Get("X-Signature")
}
if signature == "" {
    return 400, errors.New("No signature provided")
}
```

likewise for `date`.  This would also allow you to drop `rawSignature` and `rawSignDate`.